### PR TITLE
refactor: switch to a file-based DAO

### PR DESCRIPTION
**Pull request by alecthomas**
We're doing this because BoltDB has caused some Hermit binaries to time
out due to lock contention at high (>32) concurrency.

Removed "UsedAt" and consequently GC as we believe that "gc" is very
rarely used and updating usage time could be quite costly.

### CODE REVIEW

File: app/commands.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ type activated struct {
 	Validate  activatedValidateCmd `cmd:"" help:"Hermit validation." group:"global"`
 
 	Clean cleanCmd `cmd:"" help:"Clean hermit cache." group:"global"`
-	GC    gcCmd    `cmd:"" help:"Garbage collect unused Hermit packages and clean the download cache." group:"global"`
+	GC    gcCmd    `cmd:"" hidden:"" group:"global"`
```
Comment (quad) [inline app/commands.go:80]:
```suggestion
	GC    gcCmd    `cmd:"" hidden:"" group:"deprecated"`
```

Is there a way to that commands are deprecated?

**Review by quad — APPROVED** [review]
All this to avoid sqlite. 🤪

Trusting mtime… 😰😰😰

**Comment by alecthomas** [comment]
SQLite would most likely have the same issue, it is single writer too.

File: app/commands.go (hunk 1 above)
Comment (alecthomas) [reply app/commands.go:80]: No there's no real concept of it in Kong. I just hide them and write a warning.

**Comment by quad** [comment]
> SQLite would most likely have the same issue, it is single writer too.

https://sqlite.org/wal.html

**Comment by alecthomas** [comment]
> > SQLite would most likely have the same issue, it is single writer too.
> 
> https://sqlite.org/wal.html

I'm aware of how SQLite works, but it's still a single writer, so it wouldn't solve the problem we have.

> However, since there is only one WAL file, there can only be one writer at a time.

**Comment by alecthomas** [comment]
Also we can't use SQLite because we can't automate releases for M1.

**Pull request merged**